src/namespace/kv.rs

```diff
@@ -0,0 +1,167 @@
+                    Request::RequestRange(req)
+                }
+                Request::RequestPut(mut req) => {
+                    (req.key, _) = prefix_internal(&self.pfx, req.key, vec![]);
+                    Request::RequestPut(req)
+                }
+                Request::RequestDeleteRange(mut req) => {
+                    (req.key, req.range_end) = prefix_internal(&self.pfx, req.key, req.range_end);
+                    Request::RequestDeleteRange(req)
+                }
+                Request::RequestTxn(req) => Request::RequestTxn(self.prefix_txn_request(req)),
+            });
+            op
+        };
+        req.compare = req.compare.drain(..).map(prefix_cmp).collect();
+        req.success = req.success.drain(..).map(prefix_op).collect();
+        req.failure = req.failure.drain(..).map(prefix_op).collect();
+        req
+    }
+
+    fn strip_prefix_txn_response(&self, mut resp: PbTxnResponse) -> PbTxnResponse {
+        let strip_prefix_op = |mut op: ResponseOp| {
+            op.response = op.response.map(|res| match res {
+                Response::ResponseRange(r) => {
+                    Response::ResponseRange(self.strip_prefix_range_response(r))
+                }
+                Response::ResponsePut(r) => {
+                    Response::ResponsePut(self.strip_prefix_put_response(r))
+                }
+                Response::ResponseDeleteRange(r) => {
+                    Response::ResponseDeleteRange(self.strip_prefix_delete_response(r))
+                }
+                Response::ResponseTxn(r) => {
+                    Response::ResponseTxn(self.strip_prefix_txn_response(r))
+                }
+            });
+            op
+        };
+        resp.responses = resp.responses.drain(..).map(strip_prefix_op).collect();
+        resp
+    }
+
+    fn strip_prefix_put_response(&self, mut resp: PbPutResponse) -> PbPutResponse {
+        resp.prev_kv = resp.prev_kv.take().map(|mut kv| {
+            kv.key = self.strip_prefix_key(kv.key);
+            kv
+        });
+        resp
+    }
+
+    fn strip_prefix_range_response(&self, mut resp: PbRangeResponse) -> PbRangeResponse {
+        for kv in resp.kvs.iter_mut() {
+            take_mut(kv, |mut kv| {
+                kv.key = self.strip_prefix_key(kv.key);
+                kv
+            });
+        }
+        resp
+    }
+
+    fn strip_prefix_delete_response(&self, mut resp: PbDeleteResponse) -> PbDeleteResponse {
+        for kv in resp.prev_kvs.iter_mut() {
+            take_mut(kv, |mut kv| {
+                kv.key = self.strip_prefix_key(kv.key);
+                kv
+            });
+        }
+        resp
+    }
+
+    fn strip_prefix_key(&self, mut key: Vec<u8>) -> Vec<u8> {
+        debug_assert!(
+            key.starts_with(&self.pfx),
+            "{key:?} does not start with {:?}",
+            self.pfx
+        );
+        key.split_off(self.pfx.len())
```

Comment:
`split_off` has memory allocation that is not necessary for stripping prefix key.